From fac63be9e7674a0c714ca295a65e7d062b3acf7a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 23 Jun 2026 14:57:21 +0800
Subject: [PATCH 4/5] AOSCOS: data: adapt to AOSC-specfic arch. names

* Replace mips64el with loongson3.
* Replace i386 with i486.
* Replace loong64 with loongarch64.
* Add armv4, armv6hf, armv7hf.

Co-authored-by: Xinhui Yang <cyan@cyano.uk>
---
 data/cputable   | 9 ++++++---
 data/tupletable | 5 ++++-
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/data/cputable b/data/cputable
index 0da0281c3..d386fdf90 100644
--- a/data/cputable
+++ b/data/cputable
@@ -22,11 +22,14 @@ alpha		alpha		alpha.*			64	little
 amd64		x86_64		(amd64|x86_64)		64	little
 arc		arc		arc			32	little
 armeb		armeb		arm.*b			32	big
-arm		arm		arm.*			32	little
+armv4l		armv4l		armv4.*l		32	little
+armv6lhf	armv6l		armv6.*l		32	little
+armv7lhf	armv7l		armv7.*l		32	little
+arm		arm		arm[^467]*		32	little
 arm64		aarch64		aarch64			64	little
 hppa		hppa		hppa.*			32	big
-loong64		loongarch64	loongarch64		64	little
-i386		i686		(i[34567]86|pentium)	32	little
+loongarch64	loongarch64	loongarch64		64	little
+i486		i486		i[34567]86		32	little
 ia64		ia64		ia64			64	little
 m68k		m68k		m68k			32	big
 mips		mips		mips(eb)?		32	big
diff --git a/data/tupletable b/data/tupletable
index 7c306d497..8116f5aaa 100644
--- a/data/tupletable
+++ b/data/tupletable
@@ -25,7 +25,10 @@ eabi-uclibc-linux-arm		uclibc-linux-armel
 base-uclibc-linux-<cpu>		uclibc-linux-<cpu>
 eabihf-musl-linux-arm		musl-linux-armhf
 base-musl-linux-<cpu>		musl-linux-<cpu>
+eabihf-gnu-linux-armv6lhf	armv6hf
+eabihf-gnu-linux-armv7lhf	armv7hf
 eabihf-gnu-linux-arm		armhf
+eabi-gnu-linux-armv4l		armv4
 eabi-gnu-linux-arm		armel
 abin32-gnu-linux-mips64r6el	mipsn32r6el
 abin32-gnu-linux-mips64r6	mipsn32r6
@@ -33,7 +36,7 @@ abin32-gnu-linux-mips64el	mipsn32el
 abin32-gnu-linux-mips64		mipsn32
 abi64-gnu-linux-mips64r6el	mips64r6el
 abi64-gnu-linux-mips64r6	mips64r6
-abi64-gnu-linux-mips64el	mips64el
+abi64-gnu-linux-mips64el	loongson3
 abi64-gnu-linux-mips64		mips64
 x32-gnu-linux-amd64		x32
 base-gnu-linux-<cpu>		<cpu>
-- 
2.55.0

